providers/os/resources/firewalld.go

```diff
@@ -0,0 +1,1123 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package resources
+
+import (
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"io"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/spf13/afero"
+	"go.mondoo.com/cnquery/v12/llx"
+	"go.mondoo.com/cnquery/v12/providers-sdk/v1/plugin"
+	"go.mondoo.com/cnquery/v12/providers/os/connection/shared"
+	"go.mondoo.com/cnquery/v12/providers/os/resources/firewalld"
+	"go.mondoo.com/cnquery/v12/providers/os/resources/parsers"
+	"go.mondoo.com/cnquery/v12/types"
+)
+
+const defaultFirewalldConfig = "/etc/firewalld/firewalld.conf"
+
+var firewalldZoneSearchPaths = []string{
+	"/run/firewalld/zones",
+	"/etc/firewalld/zones",
+	"/usr/lib/firewalld/zones",
+	"/usr/share/firewalld/zones",
+}
+
+type mqlFirewalldInternal struct {
+	lock      sync.Mutex
+	loaded    bool
+	loadError error
+}
+
+type firewalldData struct {
```

Comment:
instead of having this additional intermediate layer, why are we not parsing the command line into the xml config structure?